### PR TITLE
Add InlinedVector::resize()

### PR DESCRIPTION
**Pull request by AspirinSJL**

### CODE REVIEW

File: src/core/lib/gprpp/inlined_vector.h [hunk 1]
```diff
@@ -125,6 +125,16 @@ class InlinedVector {
     }
   }
 
+  void resize(size_t new_size) {
+    if (new_size == size_) return;
+    while (new_size > size_) {
+      emplace_back();
+    }
+    while (new_size < size_) {
+      data()[size_-- - 1].~T();
```
Comment (markdroth) [inline src/core/lib/gprpp/inlined_vector.h]: Suggest changing `size_-- -1` to `--size_`.

Or better yet, just use `pop_back()` here.

File: test/core/gprpp/inlined_vector_test.cc [hunk 2]
```diff
@@ -486,6 +486,18 @@ TEST(InlinedVectorTest, PopBackAllocated) {
...
 
+TEST(InlinedVectorTest, Resize) {
+  const int kInlinedSize = 2;
+  InlinedVector<UniquePtr<int>, kInlinedSize> v;
+  // Size up.
+  v.resize(5);
+  EXPECT_EQ(5UL, v.size());
+  EXPECT_EQ(nullptr, v[4]);
+  // Size down.
+  v.resize(1);
```
Comment (markdroth) [inline test/core/gprpp/inlined_vector_test.cc]: Before sizing down, I suggest populating one of the elements that is going to be destroyed, to prove that the elements' dtors are properly called when this happens.

Comment (AspirinSJL) [reply test/core/gprpp/inlined_vector_test.cc]: Sorry, what does this mean?

Comment (markdroth) [reply test/core/gprpp/inlined_vector_test.cc]: I suggest adding something like the following right before this line:

```
v[4] = MakeUnique<int>(5);
```

Then, when we resize down to 1, the code should free that element.  If it doesn't, there will be an asan failure due to a memory leak.

File: src/core/lib/gprpp/inlined_vector.h (hunk 1 above)
Comment (AspirinSJL) [reply src/core/lib/gprpp/inlined_vector.h]: Done.

File: test/core/gprpp/inlined_vector_test.cc (hunk 2 above)
Comment (AspirinSJL) [reply test/core/gprpp/inlined_vector_test.cc]: I still don't understand the benefit of this change. 

Why would populating an additional element help prove the dtor is properly called during sizing down? Even without this additional element, the code should free the elements with index from 1 to 4. If it doesn't, there would already be an asan failure.

Comment (markdroth) [reply test/core/gprpp/inlined_vector_test.cc]: Currently, the elements in indexes 1 through 4 are all null, so calling the dtor does nothing.  If the dtor was not being called, this test would not fail.  Changing one of those elements to be non-null will cause the test to fail if the elements' dtor is not called.

Comment (AspirinSJL) [reply test/core/gprpp/inlined_vector_test.cc]: I see.. I was thinking about the dtor of `UniquePtr` itself.

Comment (markdroth) [reply test/core/gprpp/inlined_vector_test.cc]: It *is* the dtor of `UniquePtr<>` that we're talking about calling, but that dtor won't actually do anything that would be caught by asan if the `UniquePtr<>`'s value is null.

Comment (AspirinSJL) [reply test/core/gprpp/inlined_vector_test.cc]: Done.

Comment (AspirinSJL) [reply test/core/gprpp/inlined_vector_test.cc]: Yeah, that is what I meant to say.

**Comment by AspirinSJL** [comment]
https://github.com/grpc/grpc/issues/20519
https://github.com/grpc/grpc/issues/20436

**Pull request merged**